mlflow/spark/__init__.py

```diff
@@ -182,7 +182,60 @@ def log_model(
         registered_model_name: If given, create a model version under
             ``registered_model_name``, also creating a registered model if one
             with the given name does not exist.
-        signature: {{ signature }}
+        signature: A Model Signature object that describes the input and output Schema of the
+            model. The model signature can be inferred using `infer_signature` function
+            of `mlflow.models.signature`.
+            Note if your Spark model contains Spark ML vector type input or output column,
+            you should create ``SparkMLVector`` vector type for the column,
+            `infer_signature` function can also infer ``SparkMLVector`` vector type correctly
+            from Spark Dataframe input / output.
+            When loading a Spark ML model with ``SparkMLVector`` vector type input as MLflow
+            pyfunc model, it accepts ``Array[double]`` type input and internally converts
+            array into Spark ML vector, and then invoke Spark model ``transform`` method to do
+            inference, similarly, if the Spark ML pyfunc model has vector type output, Mlflow
+            internally converts Spark ML vector output data into ``Array[double]`` type
+            inference result.
+
+            Example:
+
+            .. code-block:: python
+
+                from mlflow.models import infer_signature
+                from pyspark.sql.functions import col
+                from pyspark.ml.classification import LogisticRegression
+                from pyspark.ml.functions import array_to_vector
+                import pandas as pd
+                import mlflow
+
+                train_df = spark.createDataFrame(
+                    [([3.0, 4.0], 0), ([5.0, 6.0], 1)], schema="features array<double>, label long"
+                ).select(array_to_vector("features").alias("features"), col("label"))
+                lor = (
+                    LogisticRegression(maxIter=2).setPredictionCol("").setProbabilityCol("prediction")
+                )
+                lor_model = lor.fit(train_df)
+
+                test_df = train_df.select("features")
+                prediction_df = lor_model.transform(train_df).select("prediction")
+
+                signature = infer_signature(test_df, prediction_df)
+
+                with mlflow.start_run() as run:
+                    mlflow.spark.log_model(
+                        lor_model,
+                        "model",
+                        signature=signature,
+                    )
+
+                model_uri = f"runs:/{run.info.run_id}/model"
+                loaded = mlflow.pyfunc.load_model(model_uri)
+
+                test_dataset = pd.DataFrame({"features": [[1.0, 2.0]]})
+
+                # `loaded.predict` accepts `Array[double]` type input column,
+                # and generates `Array[double]` type output column.
+                print(loaded.predict(test_dataset))
+
```

Comment:
```suggestion
                print(loaded.predict(test_dataset))
                # Array(...)
```
Can we add a sample output?